### ISSUE TITLE
displayReplyAllForm doesn't support avoiding pop-up blocker while partner center requests us to avoid pop-up blocker

### ISSUE THREAD

**Issue by vault79** [OfficeDev/office-js#1377]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

Although partner center request developers to avoid pop-up blocker from user interaction, `displayReplyAllForm` doesn't have way to avoid pop-up blocker.

## Expected Behavior
- As solution, I expect `displayReplyAllForm` provides feature to avoid pop-up blocker. For example, Dialog API provides [promptBeforeOpen](https://docs.microsoft.com/en-us/javascript/api/office/office.dialogoptions?view=word-js-preview#promptbeforeopen) option to avoid pop-up blocker.
- As short workaround, partner center allows app to show pop-up blocker from user interaction, if it is happened cause of office.js.

## Current Behavior
There is no way to avoid pop-up blocker when I uses `displayReplyAllForm` because of following reasons.
- In general, I should call `window.open` directly from user-interaction for avoiding popup blocker.
- For detecting and handle popup blocker, I need to  utilize returned value from `window.open`. But, I can't do this because it is wrapped in `displayReplyAllForm`.

Although I can't control this behavior, partner center decline our app with following message. So, I can't publish my app via app source.
> Customer Control 1100.5.4.2 Blocked Pop-Up Window Pop-up windows triggered by user interaction must not be blocked by browser pop-up blockers on their default settings. A pop-up blocked appears in the browsers when we click on the "insert" button within your add-in. Please update the offer to correct pop-up window behavior and re-submit your offer.

My suggestion is that providing `promptBeforeOpen` option for `displayReplyAllForm` function. I can avoid pop-up blocker when I use dialog API with this option. I want to do same thing with `displayReplyAllForm`.

## Steps to Reproduce, or Live Example
* Link to live example: Please use this [manifest file](https://contextaddinmasota.blob.core.windows.net/share3/manifest.xml)

* Additional details:
1. Open Outlook Web in Safari or Firefox
1. Load the manifest file and open add-in as Read Item mode
1. Click [Show popup] button
Popup Window Blocked message is shown in browser address bar

I directly call `displayReplyAllForm` from user interaction in the sample
```javascript
/* global Office */
import React from 'react';
import './App.css';

function App() {
  return (
    <div className="App">
      <button onClick={() => Office.context.mailbox.item.displayReplyAllForm("test")}>Show pop-up</button>
    </div>
  );
}

export default App;
```

## Context
- (Downgrade UX) This prevent user to see new reply window. So, user is confused and stop to use app.
- (Store decline) Partner center declines our app. I can't publish my app (stop my business).

## Your Environment
* Platform [PC desktop, Mac, iOS, Office on the web]: Office on the web.
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: Latest web app.
* Operating System: NA. It's not related to this bug because this bug happens in browser.
* Browser (if using Office on the web): I repro the error in Firefox and Safari.

## Useful logs
- [ ] Console errors
- [x] Screenshots
- [ ] Test file (if only happens on a particular file)

Please see this [video](https://contextaddinmasota.blob.core.windows.net/share3/pop-upblocker.mp4) as reference.

**Comment by vault79** [comment]
FYI: I'm asking question in Stack over flow, too
https://stackoverflow.com/questions/63761919/popup-blocker-handling-when-app-calls-displayreplyallform-method

**Comment by AlexJerabek** [comment]
Thanks for your question, @vault79.

@exextoc, could you please help answer this?

**Comment by exextoc** [comment]
> Although partner center request developers to avoid pop-up blocker from user interaction, `displayReplyAllForm` doesn't have way to avoid pop-up blocker.
> 
> ## Expected Behavior
> * As solution, I expect `displayReplyAllForm` provides feature to avoid pop-up blocker. For example, Dialog API provides [promptBeforeOpen](https://docs.microsoft.com/en-us/javascript/api/office/office.dialogoptions?view=word-js-preview#promptbeforeopen) option to avoid pop-up blocker.
> * As short workaround, partner center allows app to show pop-up blocker from user interaction, if it is happened cause of office.js.
> 
> ## Current Behavior
> There is no way to avoid pop-up blocker when I uses `displayReplyAllForm` because of following reasons.
> 
> * In general, I should call `window.open` directly from user-interaction for avoiding popup blocker.
> * For detecting and handle popup blocker, I need to  utilize returned value from `window.open`. But, I can't do this because it is wrapped in `displayReplyAllForm`.
> 
> Although I can't control this behavior, partner center decline our app with following message. So, I can't publish my app via app source.
> 
> > Customer Control 1100.5.4.2 Blocked Pop-Up Window Pop-up windows triggered by user interaction must not be blocked by browser pop-up blockers on their default settings. A pop-up blocked appears in the browsers when we click on the "insert" button within your add-in. Please update the offer to correct pop-up window behavior and re-submit your offer.
> 
> My suggestion is that providing `promptBeforeOpen` option for `displayReplyAllForm` function. I can avoid pop-up blocker when I use dialog API with this option. I want to do same thing with `displayReplyAllForm`.
> 
> ## Steps to Reproduce, or Live Example
> * Link to live example: Please use this [manifest file](https://contextaddinmasota.blob.core.windows.net/share3/manifest.xml)
> * Additional details:
> 
> 1. Open Outlook Web in Safari or Firefox
> 2. Load the manifest file and open add-in as Read Item mode
> 3. Click [Show popup] button
>    Popup Window Blocked message is shown in browser address bar
> 
> I directly call `displayReplyAllForm` from user interaction in the sample
> 
> ```js
> /* global Office */
> import React from 'react';
> import './App.css';
> 
> function App() {
>   return (
>     <div className="App">
>       <button onClick={() => Office.context.mailbox.item.displayReplyAllForm("test")}>Show pop-up</button>
>     </div>
>   );
> }
> 
> export default App;
> ```
> 
> ## Context
> * (Downgrade UX) This prevent user to see new reply window. So, user is confused and stop to use app.
> * (Store decline) Partner center declines our app. I can't publish my app (stop my business).
> 
> ## Your Environment
> * Platform [PC desktop, Mac, iOS, Office on the web]: Office on the web.
> * Host [Excel, Word, PowerPoint, etc.]: Outlook
> * Office version number: Latest web app.
> * Operating System: NA. It's not related to this bug because this bug happens in browser.
> * Browser (if using Office on the web): I repro the error in Firefox and Safari.
> 
> ## Useful logs
> * [ ]  Console errors
> * [x]  Screenshots
> * [ ]  Test file (if only happens on a particular file)
> 
> Please see this [video](https://contextaddinmasota.blob.core.windows.net/share3/pop-upblocker.mp4) as reference.

Thank you for reporting this issue. Yes we could repro this issue. We have added this to our backlog; however we don't have any timelines to share for the same.

**Comment by vault79** [comment]
@AlexJerabek Thank you for your quick triage and dispatch.

@exextoc Thank you for checking repro quickly. I'm OK that we don't have any timelines to fix it, but would you please communicate with partner center team internally to allow app which shows pop-up blocker by this issue ? It blocks our app submission...

**Comment by vault79** [comment]
FYI for other developers who has same issue and certification trouble.

For resolving my business blocker, I re-submit my app to partner center with this issue url. Partner center advices me to submit support ticket from here.
https://aka.ms/marketplacepublishersupport 

If you encounter same issue, please refer this info. I'll update if I have progress.
